src/BookNavigator/search/a-search-result.js

```diff
@@ -12,17 +11,44 @@ export class BookSearchResult extends LitElement {
   constructor() {
     super();
 
-    this.matchRegex = new RegExp('{{{(.+?)}}}', 'g');
+    this.matchRegex = new RegExp('{{{(.+?)}}}', 'gs');
   }
 
   createRenderRoot() {
     return this;
   }
 
+  /**
+   * Converts the search hit to a `<p>` template containing the full search result with all of
+   * its `{{{triple-brace-delimited}}}` matches replaced by `<mark>HTML mark tags</mark>`.
+   *
+   * This approach safely avoids the use of `unsafeHTML` and leaves any existing HTML tags
+   * in the snippets intact (as inert text), rather than stripping them away with DOMPurify.
+   */
   highlightedHit(hit) {
-    return html`
-      <p>${unsafeHTML(hit.replace(this.matchRegex, '<mark>$1</mark>'))}</p>
-    `;
+    const matches = hit.matchAll(this.matchRegex);
+    const templates = [];
+
+    // Convert each match into an HTML template that includes:
+    //  - Everything from the end of the previous match (or the beginning of the
+    //      string) up to the current match, as raw text.
+    //  - The current match (excluding the curly braces) wrapped in a `<mark>` tag.
+    let index = 0;
+    for (const match of matches) {
+      if (match.index != null) {
+        templates.push(html`
+          ${hit.slice(index, match.index)}
+          <mark>${match[1]}</mark>
+        `);
+        index = match.index + match[0].length;
+      }
+    }
+
+    // Include any text from the last match to the end
+    templates.push(html`${hit.slice(index)}`);
+
+    // Squash everything into a single p template
+    return html`<p>${templates}</p>`;
```

Comment:
A simpler approach might be something like:

```js
return html`<p>${unsafeHTML(escapeHTML(hit).replace(/{{{/g, '<mark>').replace(/}}}/g, '</mark>'))}</p>`
```

escapeHTML exists in our utils :)

Reply:
Ah, figures we have an escaper for this already! Thanks for the suggestion!